Server/src/controllers/getDatasetController.ts

```diff
@@ -0,0 +1,110 @@
+     * then return a response with the array of data sets. If the request is invalid, or some other problem arrises, 
+     * it will throw an error.
+     * 
+     * @param request
+     * An object containing the request information and parameters: Request 
+     * @param response 
+     * An object containing a response: Response
+     */
+    async createRequestForData(request: Request, response: Response) {
+        let validateData = this.validateInputData(request)
+        if (!validateData) {
+            response.status(500).send("Invalid search params entered");
+        }
+        else {
+            let requestParams: any = { ...request.query };
+            this.processedRequest = requestParams;
+            try {
+                const retrieveDataObject = new retrieveData();
+                let arrayOfData = await retrieveDataObject.getArrayOfDatasets(this.processedRequest)
+                return response.status(200).send(arrayOfData);
+            } catch (err) {
+                response.status(500).send(err);
+            }
+        }
+    }
+
+    /**
+     * This controller will take a request, grab the user ID, and if it a real number will then process the request,
+     * send it to the getDataService to acquire an array of data sets that were uploaded by this user ID, and
+     * then return a response with the array of data sets. If the request is invalid, or some other problem arrises, 
+     * it will throw an error.
+     * 
+     * @param request
+     * An object containing the request information and parameters: Request 
+     * @param response 
+     * An object containing a response: Response
+     */
+    async createRequestForUserUploads(request: Request, response: Response) {
+        let requestParam = request.params.uploadedBy;
+        let userId: number = +requestParam;
+        if (isNaN(userId)) {
+            response.status(500).send("Invalid search params entered");
+        }
+        else {
+            try {
+                const retrieveDataObject = new retrieveData();
+                let arrayOfData = await retrieveDataObject.getUserUploadedDatasets(userId)
+                return response.status(200).send(arrayOfData);
+            } catch (err) {
+                response.status(500).send(err);
+            }
+        }
+    }
+
+    /**
+     * This controller will take a request, grab the user ID, and if it a real number will then process the request,
+     * send it to the getDataService to acquire an array of data sets that were favorited by this user ID, and
+     * then return a response with the array of data sets. If the request is invalid, or some other problem arrises, 
+     * it will throw an error.
+     * 
+     * @param request
+     * An object containing the request information and parameters: Request 
+     * @param response 
+     * An object containing a response: Response
+     */
+    async createRequestForUserFavorites(request: Request, response: Response) {

```

Comment:
What is the difference here between userFavorites and userUploads?

Reply:
One is referring to data sets that a user wants to keep track of and the other is specifically _who_ uploaded a particular data set

Reply:
As he mentioned, the userUploads is used for the Uploaded table which shows the user all the datasets they uploaded to the database. Versus a table referencing all the datasets from the database he wants to keep an eye on or have easy access to.

Reply:
Can we update the names in that case. Its confusing to understand the difference without this explanation

Reply:
Names have been changed to userUploadedDatasets and userSavedDatsets for better clarity